tests/integration-tests/tests/update/test_update.py

```diff
@@ -439,3 +441,144 @@ def get_batch_spot_bid_percentage(stack_name, region):
+    init_config_file = pcluster_config_reader(global_custom_ami=pcluster_ami_id)
+    cluster = clusters_factory(init_config_file)
+    remote_command_executor = RemoteCommandExecutor(cluster)
+    scheduler_commands = scheduler_commands_factory(remote_command_executor)
+    _check_queue_ami(cluster, ec2, pcluster_ami_id, "queue1")
+    _check_queue_ami(cluster, ec2, pcluster_ami_id, "queue2")
+
+    _test_queue_strategy_without_running_job(
+        pcluster_config_reader, pcluster_ami_id, cluster, region, os, remote_command_executor, scheduler_commands
+    )
+
+    queue1_job_id = scheduler_commands.submit_command_and_assert_job_accepted(
+        submit_command_args={
+            "command": "sleep 3000",
+            "nodes": -1,
+            "partition": "queue1",
+            "other_options": "-a 1-5",  # instance type has 4 cpus per node, which requre 2 nodes to run the job
+        }
+    )
+
+    queue2_job_id = scheduler_commands.submit_command_and_assert_job_accepted(
+        submit_command_args={"command": "sleep 3000", "nodes": -1, "partition": "queue2", "other_options": "-a 1-5"}
+    )
+    # Wait for the job to run
+    scheduler_commands.wait_job_running(queue1_job_id)
+    scheduler_commands.wait_job_running(queue2_job_id)
+
+    pcluster_copy_ami_id = ami_copy(
+        pcluster_ami_id, "-".join(["test", "update", "computenode", generate_random_string()])
+    )
+    # test update without setting queue strategy, update will fail
+    _test_update_without_queue_strategy(pcluster_config_reader, pcluster_ami_id, pcluster_copy_ami_id, cluster)
+
+    updated_config_file = pcluster_config_reader(
+        config_file="pcluster.config.update_drain.yaml",
+        global_custom_ami=pcluster_ami_id,
+        custom_ami=pcluster_copy_ami_id,
+    )
+    cluster.update(str(updated_config_file))
+    # check chef client log contains expected log
+    assert_errors_in_logs(
+        remote_command_executor,
+        ["/var/log/chef-client.log"],
+        ["Queue update strategy is \\(DRAIN\\)", "Adding queue \\(queue2\\) to list of queue to be updated"],
+    )
+
+    # after cluster update, check if jobs are still in running state
+    scheduler_commands.assert_job_state(queue1_job_id, "RUNNING")
+    scheduler_commands.assert_job_state(queue2_job_id, "RUNNING")
+    # assert queue1 node state are in working state
+    queue1_nodes = scheduler_commands.get_compute_nodes("queue1")
+    assert_compute_node_states(scheduler_commands, queue1_nodes, expected_states=["mixed", "allocated"])
+    # assert queue2 node state are in draining status
+    queue2_nodes = scheduler_commands.get_compute_nodes("queue2")
+    assert_compute_node_states(scheduler_commands, queue2_nodes, expected_states=["draining", "draining!"])
+    # retrieve queue2 instance again, check ami, they are not replaced
+    _check_queue_ami(cluster, ec2, pcluster_ami_id, "queue1")
+    _check_queue_ami(cluster, ec2, pcluster_ami_id, "queue2")
+
+    # cancel job in queue2
```

Comment:
instead of ´cancel + resubmit` you can just `requeue` the job, with `scontrol requeue <job_id>`